test/e2e/charts/vault/templates/vault-deployment.yaml

```diff
@@ -37,9 +37,12 @@ spec:
             cpu: 200m
             memory: 256Mi
         securityContext:
-         capabilities:
-           add:
-             - IPC_LOCK
+          runAsNonRoot: true
+          runAsGroup: 1000
+          runAsUser: 100
+          capabilities:
+            add:
+              - IPC_LOCK
```

Comment:
This Vault chart is home made, but there is now an upstream chart which sets these by default: https://github.com/hashicorp/vault-helm/pull/80

Perhaps we should just use that chart in our tests.

Reply:
vault calls `mlock` to prevent memory containing keys being swapped to disk which means it needs this capability

Reply:
I ended up reverting these changes because the tests failed anyway with the new image and rbac config. See https://github.com/jetstack/cert-manager/pull/4044 where I extracted this change in to a separate PR.